### PR TITLE
Add basic telemetry infrastructure

### PR DESCRIPTION
**Pull request by puehringer**
Adds basic telemetry using opentelemetry, prometheus, tempo, loki, grafana, ...

## Configuration

For local testing, start the corresponding services via: 
```bash
docker compose -f ./deploy/telemetry/docker-compose.telemetry.yml up
```

and set the following env variables: 

```
VISYN_CORE__TELEMETRY__ENABLED=true
VISYN_CORE__TELEMETRY__SERVICE_NAME=my_app.local
VISYN_CORE__TELEMETRY__GLOBAL_EXPORTER__ENDPOINT=http://localhost:4318/
# VISYN_CORE__TELEMETRY__GLOBAL_EXPORTER__HEADERS='{"Authorization": "Basic MySecret:)"}'
```

## Screenshots
image

Browse available trace ids and inspect one in detail:

grafik

Inspecting a trace in detail with the flame chart:

grafik

## Todos

- [ ] Checkout https://github.com/autometrics-dev/autometrics-py/

### CODE REVIEW

**Comment by mstreit** [comment]
Oh, really cool!

File: requirements_extras_telemetry.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+opentelemetry-exporter-otlp==1.15.0
+opentelemetry-exporter-prometheus==1.12.0rc1
+opentelemetry-instrumentation-fastapi==0.36b0
+opentelemetry-instrumentation-httpx==0.36b0
+opentelemetry-instrumentation-logging==0.36b0
+opentelemetry-instrumentation-requests==0.36b0
+opentelemetry-instrumentation-sqlalchemy==0.36b0
+opentelemetry-instrumentation-system-metrics==0.36b0
+opentelemetry-sdk==1.15.0
+python-logging-loki==0.3.1
```
Comment (thinkh) [inline requirements_extras_telemetry.txt:11]: Update versions in requirements.txt

**Comment by thinkh** [comment]
https://www.dynatrace.com/support/help/extend-dynatrace/opentelemetry/opentelemetry-traces/opentelemetry-ingest/opent-python#tabgroup--instrument-application--without-oneagent

File: visyn_core/settings/model.py [hunk 2]
```diff
@@ -65,11 +73,75 @@ class SecuritySettings(BaseModel):
...
+class LogsTelemetrySettings(BaseTelemetrySettings):
+    exporter: LogsExporterTelemetrySettings | None = None
+
+
+class TelemetrySettings(BaseModel):
+    enabled: bool = False
+    """
+    Globally enable or disable telemetry.
+    """
+    service_name: str
```
Comment (thinkh) [inline visyn_core/settings/model.py:125]:
```suggestion
    service_name: str
    """
    Service name must be full qualified name (e.g., myapp.app.datavisyn.io)
    """
```

File: visyn_core/middleware/fastapi_metrics_middleware.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    name="fastapi_requests_in_progress", description="Gauge of requests by method and path currently being processed", unit="1"
+)
+users_counter = meter.create_counter(name="fastapi_users_total", description="Total count of unique users", unit="1")
+exceptions_counter = meter.create_counter(
+    name="fastapi_exceptions_total", description="Total count of exceptions raised by path and exception type", unit="1"
+)
+requests_in_progress = meter.create_observable_gauge(name="requests", description="number of requests", unit="1")
+
+
+# TODO: This middleware is quite slow, it should be replaced with one similar to the ASGI instrumentor middleware
```
Comment (thinkh) [inline visyn_core/middleware/fastapi_metrics_middleware.py:36]: Is this something we should address now or do you want to create a follow-up issue?

File: visyn_core/settings/model.py [hunk 4]
```diff
@@ -46,6 +46,12 @@ class AlbSecurityStoreSettings(BaseModel):
     signout_url: str | None = None
 
 
+class RandomGeneratedSecurityStoreSettings(BaseModel):
+    enable: bool = False
+    # TODO: Have a global datadir settings in core and extend it here.
+    file: str = "./fakeUsers.db"
```
Comment (thinkh) [inline visyn_core/settings/model.py:52]: What about this TODO?

File: visyn_core/tests/test_telemetry.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+from fastapi.testclient import TestClient
+from prometheus_client.parser import text_string_to_metric_families
+
+
+@pytest.mark.parametrize(
+    "workspace_config",
+    [
+        {
+            "tdp_core": {
+                "enabled_plugins": ["tdp_core"],
```
Comment (thinkh) [inline visyn_core/tests/test_telemetry.py:11]:
```suggestion
            "visyn_core": {
                "enabled_plugins": ["visyn_core"],
```

Is it really still tdp_core?

**Review by thinkh — CHANGES_REQUESTED** [review]
Please find my remarks below. I'd also suggest to add a Readme.md with the basic instructions how to set up and run the telemetry for an app and also for a local Docker setup.

File: visyn_core/settings/model.py (hunk 4 above)
Comment (puehringer) [reply visyn_core/settings/model.py:52]: This should have never been in this PR, was added by mistake. Will remove it.

File: visyn_core/tests/test_telemetry.py (hunk 5 above)
Comment (puehringer) [reply visyn_core/tests/test_telemetry.py:11]: Both works (it falls back to tdp_core if visyn_core key is not found), but more correct is visyn_core. Will change it.

File: visyn_core/middleware/fastapi_metrics_middleware.py (hunk 3 above)
Comment (puehringer) [reply visyn_core/middleware/fastapi_metrics_middleware.py:36]: It's still very fast, but FastAPI/starlette HTTPMiddleware is slower than the "raw" middleware. Nothing to be fixed however, only when it becomes a problem (which it isn't yet).

**Review by thinkh — APPROVED** [review]
Thanks for integrating the feedback. Looks good to me. 👍

**Pull request merged**